### PR TITLE
cant not archive a validated enrollment

### PR DESCRIPTION
**Pull request by Isalafont**
instructor and administrator can no longer archive a validated enrollment. enrollment has to be revoked before.

https://user-images.githubusercontent.com/43042737/229558005-e373ddca-02bb-4dc5-aa4c-8c501192d892.mov

### CODE REVIEW

**Comment by Isalafont** [comment]
[ETQ I, je ne peux pas archiver une habilitation validée (besoin de la révoquer avant)](https://trello.com/c/frgUiwpn/1774-etq-i-je-ne-peux-pas-archiver-une-habilitation-valid%C3%A9e-besoin-de-la-r%C3%A9voquer-avant)

File: backend/app/policies/enrollment_policy.rb [hunk 1]
```diff
@@ -77,8 +77,8 @@ def archive?
       user.belongs_to_organization?(record) &&
       user.is_demandeur?(record)
 
-    instructor_right = user.is_instructor?(record.target_api)
-    administrator_right = user.is_administrator?
+    instructor_right = !record.status_validated? && user.is_instructor?(record.target_api)
+    administrator_right = !record.status_validated? && user.is_administrator?
```
Comment (pierreericgarcia) [inline backend/app/policies/enrollment_policy.rb:81]: I don't think these conditions are useful as `record.can_archive_status?` returns already `false` if we're trying to transition from a `validated` state to an `archived` state.

**Review by pierreericgarcia — CHANGES_REQUESTED** [review]

File: backend/app/policies/enrollment_policy.rb (hunk 1 above)
Comment (skelz0r) [reply backend/app/policies/enrollment_policy.rb:81]: Indeed, `record.can_archive_status?` already check this behaviour.

**Review by skelz0r — CHANGES_REQUESTED** [review]
cf my only com

**Pull request merged**